pandas/io/excel/_base.py

```diff
@@ -358,6 +358,10 @@ def read_excel(
     mangle_dupe_cols=True,
     storage_options: StorageOptions = None,
 ):
+    kwargs = locals().copy()
+    for each in kwargs:
+        if isinstance(locals()[each], dict):
+            kwargs[each] = locals()[each].copy()
```

Comment:
This is essentially trying to do a deepcopy?

Reply:
Hi @mzeitlin11 . Yes it is. I tried to use deepcopy first, but it was failing multiple tests.

I got the idea of creating a copy of kwargs from read_csv (which as you mentioned doesn't produce this unnecessary side effect). So I thought it would be a good idea to keep it consistent.

https://github.com/pandas-dev/pandas/blob/d60e6870f9c69d3407a5edb2298cb2181bc0bb83/pandas/io/parsers/readers.py#L566-L586

Reply:
> I got the idea of creating a copy of kwargs from read_csv (which as you mentioned doesn't produce this unnecessary side effect). So I thought it would be a good idea to keep it consistent.

Makes sense.  That case is a bit different though because `kwargs` itself was being modified. Here the issue is limited to the `dtypes` dict.  If there's a clean way to copy (or not modify in place) the dtypes dict, I _think_ that would be cleaner.

Reply:
Sure. I will work along those lines. Thanks!

Reply:
Let me know if you run into any issues!